pkg/proxy/proxy.go

```diff
@@ -116,6 +172,10 @@ func (h *Proxy) doModifyResponse(config TargetConfig) func(*http.Response) error
 
 			return errors.New("server error")
 		default:
+			if err := h.doJSONRPCValidation(resp); err != nil {
+				return errors.Wrap(err, "json rpc validation failed")
+			}
```

Comment:
Nice. This will also surface the problem when AWS drops the packets because of EC2 network allowance.